contracts/RelaySugar.vy

```diff
@@ -54,61 +54,136 @@ interface IAutoCompounder:
 
 # Vars
 factory: public(IAutoCompounderFactory)
-ve_sugar: public(IVeSugar)
+voter: public(IVoter)
 ve: public(IVotingEscrow)
+token: public(address)
 
 @external
-def __init__(_factory: address, _ve_sugar: address, _ve: address):
+def __init__(_factory: address, _voter: address):
   """
   @dev Set up our external factory contract
   """
   self.factory = IAutoCompounderFactory(_factory)
-  self.ve_sugar = IVeSugar(_ve_sugar)
-  self.ve = IVotingEscrow(_ve)
+  self.voter = IVoter(_voter)
+  self.ve = IVotingEscrow(self.voter.ve())
+  self.token = self.ve.token()
 
 @external
 @view
-def all() -> DynArray[Relay, MAX_COMPOUNDERS]:
+def all(_account: address) -> DynArray[Relay, MAX_COMPOUNDERS]:
+  """
+  @notice Returns all Relays and account's deposits
+  @return Array of Relay structs
+  """
+  autocompounders: DynArray[Relay, MAX_COMPOUNDERS] = self._autocompounders(_account)
+
+  return autocompounders
+
+@internal
+@view
+def _autocompounders(_account: address) -> DynArray[Relay, MAX_COMPOUNDERS]:
   """
-  @notice Returns all AutoCompounders
-  @return Array of AutoCompounder structs
+  @notice Returns all Relays and account's deposits
+  @return Array of Relay structs
   """
   compounders: DynArray[Relay, MAX_COMPOUNDERS] = empty(DynArray[Relay, MAX_COMPOUNDERS])
   addresses: DynArray[address, MAX_COMPOUNDERS] = self.factory.autoCompounders()
-  
-  for index in range(0, len(addresses)):
+
+  for index in range(0, MAX_COMPOUNDERS):
+    if index == len(addresses):
+      break
+
     autocompounder: IAutoCompounder = IAutoCompounder(addresses[index])
     managed_id: uint256 = autocompounder.tokenId()
-    managed_nft: VeNFT = self.ve_sugar.byId(managed_id)
-    inactive: bool = self.ve.deactivated(managed_id)
-
-    compounders.append(Relay({
-      name: autocompounder.name(),
-      managed_nft: managed_nft,
-      address: addresses[index],
-      inactive: inactive
-    }))
+
+    relay: Relay = self._byId(managed_id, _account)
```

Comment:
optimization: looks like we can just pass the `autocompounder.address` to the `_byId(...)` directly and remove extra calls in the internal function?